python-checks/src/main/java/org/sonar/python/checks/hotspots/CorsCheck.java

```diff
@@ -0,0 +1,356 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.hotspots;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.ArgList;
+import org.sonar.plugins.python.api.tree.Argument;
+import org.sonar.plugins.python.api.tree.AssignmentStatement;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonar.plugins.python.api.tree.Decorator;
+import org.sonar.plugins.python.api.tree.DictionaryLiteral;
+import org.sonar.plugins.python.api.tree.DictionaryLiteralElement;
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.ExpressionList;
+import org.sonar.plugins.python.api.tree.KeyValuePair;
+import org.sonar.plugins.python.api.tree.ListLiteral;
+import org.sonar.plugins.python.api.tree.Name;
+import org.sonar.plugins.python.api.tree.RegularArgument;
+import org.sonar.plugins.python.api.tree.StringLiteral;
+import org.sonar.plugins.python.api.tree.SubscriptionExpression;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.plugins.python.api.tree.Tuple;
+
+import static org.sonar.plugins.python.api.tree.Tree.Kind.ASSIGNMENT_STMT;
+import static org.sonar.plugins.python.api.tree.Tree.Kind.CALL_EXPR;
+import static org.sonar.plugins.python.api.tree.Tree.Kind.DECORATOR;
+import static org.sonar.plugins.python.api.tree.Tree.Kind.DICTIONARY_LITERAL;
+import static org.sonar.plugins.python.api.tree.Tree.Kind.KEY_VALUE_PAIR;
+import static org.sonar.plugins.python.api.tree.Tree.Kind.LIST_LITERAL;
+import static org.sonar.plugins.python.api.tree.Tree.Kind.NAME;
+import static org.sonar.plugins.python.api.tree.Tree.Kind.REGULAR_ARGUMENT;
+import static org.sonar.plugins.python.api.tree.Tree.Kind.STRING_LITERAL;
+import static org.sonar.plugins.python.api.tree.Tree.Kind.SUBSCRIPTION;
+import static org.sonar.plugins.python.api.tree.Tree.Kind.TUPLE;
+import static org.sonar.python.semantic.SymbolUtils.getTypeName;
+
+@Rule(key = "S5122")
+public class CorsCheck extends PythonSubscriptionCheck {
+
+  private static final String MESSAGE = "Make sure this permissive CORS policy is safe here.";
+
+  private static final String DJANGO_ALLOW_ALL = "CORS_ORIGIN_ALLOW_ALL";
+  private static final String DJANGO_WHITELIST = "CORS_ORIGIN_REGEX_WHITELIST";
+
+  private static final String STAR = "*";
+
+  private static final List<String> REGEX_TO_REPORT = Arrays.asList(
+    ".*",
+    ".+",
+    "^.*$",
+    "^.+$",
+    STAR
+  );
```

Comment:
Would it be possible and only keep `*`, `.*` and `.+` in the list and and trim '$' and '^' characters before matching? It would cover permissive regex like `^.+`.